src/osd/PeeringState.cc

```diff
@@ -5653,6 +5656,11 @@ boost::statechart::result PeeringState::Active::react(const MNotifyRec& notevt)
       ps->discover_all_missing(
 	context<PeeringMachine>().get_recovery_ctx().msgs);
     }
+    // check if it is a previous down acting member that's coming back.
+    // if so, request pg_temp change to trigger a new interval transition
+    pg_shard_t auth_log_shard;
+    bool history_les_bound = false;
+    ps->choose_acting(auth_log_shard, false, &history_les_bound, true);
```

Comment:
Add a debugging statement here at level 10.

Reply:
added and the case is hit with following log:
```
2020-02-19 16:44:26.822991 7f846caf7700 10 osd.30 pg_epoch: 73119 pg[1.3e6( v 71749'838968 (71624'837389,71749'838968] local-lis/les=73117/73118 n=1481 ec=107/107 lis/c 73117/73093 les/c/f 73118/73094/0 73096/73117/70738) [30,17,33]/[30,33] backfill=[17] r=0 lpr=73117 pi=[73093,73117)/1 rops=1 bft=17 crt=71749'838968 lcod 71749'838967 mlcod 0'0 active+undersized+degraded+remapped+backfilling mbc={255={}}] state<Started/Primary/Active>: Active: got notify from previous acting member 63 , requesting pg_temp change
2020-02-19 16:44:26.815028 7f52417b5700 10 osd.38 pg_epoch: 73119 pg[1.3f1( v 71749'884636 (71624'883060,71749'884636] local-lis/les=73117/73118 n=1570 ec=107/107 lis/c 73117/73093 les/c/f 73118/73094/0 73096/73117/73071) [17,38,58]/[38,58] backfill=[17] r=0 lpr=73117 pi=[73093,73117)/1 rops=1 bft=17 crt=71749'884636 lcod 71746'884635 mlcod 0'0 active+undersized+degraded+remapped+backfilling mbc={255={}}] state<Started/Primary/Active>: Active: got notify from previous acting member 63 , requesting pg_temp change
```

Reply:
added and the case is hit with following log:
```
2020-02-19 16:44:26.822991 7f846caf7700 10 osd.0 pg_epoch: 73119 pg[1.3e( v 71749'838968 (71624'837389,71749'838968] local-lis/les=73117/73118 n=1481 ec=107/107 lis/c 73117/73093 les/c/f 73118/73094/0 73096/73117/70738) [0,17,33]/[0,33] backfill=[17] r=0 lpr=73117 pi=[73093,73117)/1 rops=1 bft=17 crt=71749'838968 lcod 71749'838967 mlcod 0'0 active+undersized+degraded+remapped+backfilling mbc={255={}}] state<Started/Primary/Active>: Active: got notify from previous acting member 3 , requesting pg_temp change
```

Reply:
Did you observe this in a teuthology run?  This is a pretty core change to peering, and it's important that it have coverage in our regular test runs, both because we need to ensure we don't rebreak it and because we want some confidence that it won't break something else.

Reply:
Nope, but I managed to add a reproducer anyway.